src/main/java/org/zalando/nakadi/service/subscription/SubscriptionValidationService.java

```diff
@@ -100,7 +100,7 @@ public void validateSubscriptionChange(final Subscription old, final Subscriptio
         if (!Objects.equals(newValue.getInitialCursors(), old.getInitialCursors())) {
             throw new SubscriptionUpdateConflictException("Not allowed to change initial cursors");
         }
-        authorizationValidator.validateAuthorization(old.getAuthorization(), newValue.getAuthorization());
+        authorizationValidator.validateAuthorization(old.asResource(), newValue.asBaseResource());
```

Comment:
As you call `newValue.asBaseResource()` - it will create a resource without a name. I think it's fine to create a resource without a name for a subscription that is not created yet, but in this user modifies subscription that already exists so I think the id should be used as a resource name.

Reply:
Resolved